frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -134,65 +134,63 @@ export const StyledStreamlitMarkdown =
     }
   )
 
-export const StyledLinkIconContainer = styled.div(() => ({
-  position: "relative",
-  left: "calc(-2.5rem - 0.5rem)",
-  width: "calc(100% + 2.5rem + 0.5rem)",
-  display: "flex",
-  alignItems: "center",
-  overflow: "visible",
+export const StyledHeadingWithActionElements = styled.div(({ theme }) => ({
+  "h1, h2, h3, h4, h5, h6, span": {
+    scrollMarginTop: theme.spacing.threeXL,
+  },
+  ...sharedMarkdownStyle(theme),
+
+  // break-word & pretty makes most headings break in a nicer way than break-all while still
+  // preventing overflowing of the container to the side. Long headings without whitespaces or hyphens might still look weird
+  wordBreak: "break-word",
+  textWrap: "pretty",
+
+  // show link-icon when hovering somewhere over the heading
+  // we use opacity instead of visibility because the react-testing-library seems
+  // to have issues with queryByRole('link') otherwise, even when triggering hover-events
+  "& .stHeaderActionElements > a": {
+    visibility: "hidden",
+  },
   ":hover": {
-    a: {
-      opacity: 1,
-      transform: "scale(1)",
-      transition: "none",
+    "& .stHeaderActionElements > a": {
+      visibility: "visible",
     },
   },
 }))
 
-export const StyledLinkIcon = styled.a(({ theme }) => ({
-  position: "absolute",
-  marginRight: "0.5rem",
+export const StyledHeadingActionElements = styled.span(({ theme }) => ({
+  marginLeft: "0.5rem",
+  display: "inline-flex",
+  gap: "0.5rem",
+
+  verticalAlign: "middle",
 
+  "& > *": {
+    // make sure that the elements are hoverable and are not "covered" by margins etc. of other elements
+    zIndex: theme.zIndices.sidebar + 1,
```

Comment:
Does this also work in dialog?

Reply:
I have removed the zIndex, I think it was there before to show the icon over the sidebar when it was overlapping. I have added an e2e test to ensure that hovering works in a dialog